tasks/connectors/qualys_was/qualys_was.rb

```diff
@@ -230,7 +235,14 @@ def domain_detail(find_from)
       end
 
       def name(find_from)
-        "#{find_from['qid']}-#{find_from['name']}"
+        if @options[:match_ScannerIdentifier_with_vuln_def_name]
+          return if find_from.nil?
+
+          structured_name = [find_from['qid'], find_from['id'], find_from['name']].compact.join('-')
+          structured_name unless structured_name.empty?
```

Comment:
I'm not sure this is necessary, I'd expect at least finding ID to be present for all findings
If QID/ID/Name were simultaneously resulting in nil values it could indicate a larger issue and we might want to catch that failure, if not, we'd still want to return something in here

Reply:
discussion above :D